apis/r/tests/testthat/test-shape.R

```diff
@@ -56,6 +59,60 @@ test_that("SOMADataFrame shape", {
       expect_true(is.na(sjid_maxshape))
     }
 
+    dom = sdf$domain()
+    mxd = sdf$maxdomain()
+
+    # First check names
+    if (case_name == "one") {
+      expect_equal(names(dom), c("soma_joinid"))
+      expect_equal(names(mxd), c("soma_joinid"))
+
+    } else if (case_name == "two") {
+      expect_equal(names(dom), c("soma_joinid", "foo"))
+      expect_equal(names(mxd), c("soma_joinid", "foo"))
+
+    } else if (case_name == "three") {
+      expect_equal(names(dom), c("soma_joinid", "baz"))
+      expect_equal(names(mxd), c("soma_joinid", "baz"))
+
+    } else if (case_name == "four") {
+      expect_equal(names(dom), c("baz", "foo"))
+      expect_equal(names(mxd), c("baz", "foo"))
+    }
+
+    # Then check all slots are pairs
+    for (name in names(dom)) {
+      expect_equal(length(dom[[name]]), 2)
+    }
+    for (name in names(mxd)) {
+      expect_equal(length(mxd[[name]]), 2)
+    }
```

Comment:
These test modifications can be simplified:
 - we've already determined that the names of `dom` and `mxd` are identical; we can iterate through them both simultaneously
 - we can use `expec_length()` instead of `expect_equal()` for more informative test failure messages
```suggestion
    for (name in names(dom)) {
      expect_length(dom[[name]], 2L)
      expect_length(mxd[[name]], 2L)
    }
```